src/main/java/uk/co/ramp/statistics/StatisticsRecorderImpl.java

```diff
@@ -0,0 +1,97 @@
+package uk.co.ramp.statistics;
+
+import java.util.*;
+import uk.co.ramp.io.types.StandardProperties;
+import uk.co.ramp.people.AlertStatus;
+import uk.co.ramp.people.Case;
+import uk.co.ramp.statistics.types.ImmutableInfection;
+import uk.co.ramp.statistics.types.ImmutableRValueOutput;
+import uk.co.ramp.statistics.types.Infection;
+
+public class StatisticsRecorderImpl implements StatisticsRecorder {
+
+  private final Map<Integer, Integer> personDaysIsolation = new HashMap<>();
+  private final Map<Integer, Integer> peopleInfected = new HashMap<>();
+  private final Map<Integer, Integer> contactsTraced = new HashMap<>();
+  private final Map<Integer, List<Infection>> r0Progression = new HashMap<>();
+  private final Map<AlertStatus, Integer> incorrectTests = new EnumMap<>(AlertStatus.class);
+
+  private final StandardProperties properties;
+
+  public StatisticsRecorderImpl(StandardProperties properties) {
+    this.properties = properties;
+  }
+
+  public Map<Integer, Integer> getContactsTraced() {
+    return contactsTraced;
+  }
+
+  public Map<Integer, Integer> getPeopleInfected() {
+    return peopleInfected;
+  }
+
+  public Map<Integer, Integer> getPersonDaysIsolation() {
+    return personDaysIsolation;
+  }
+
+  public Map<Integer, List<Infection>> getR0Progression() {
+    return r0Progression;
+  }
+
+  @Override
+  public int getFalseNegatives() {
+    return incorrectTests.get(AlertStatus.TESTED_NEGATIVE);
+  }
+
+  @Override
+  public int getFalsePositives() {
+    return incorrectTests.get(AlertStatus.TESTED_POSITIVE);
```

Comment:
This will throw a NPE when TESTED_POSITIVE is not in the map.
```suggestion
    return incorrectTests.getOrDefault(AlertStatus.TESTED_POSITIVE, 0);
```